internal/env/supported_configurations.go

```diff
@@ -0,0 +1,88 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+package env
```

Comment:
nit: should these packages (this one and the one in instrumentation/)  just be named `config`? since we are already reading from the env and a config file.

Also we could extend this package in the future with some config specific features we have scattered around in the repo (like some sort of caching for performance so the config is only read from here, telemetry tracking, type parsing to bool, slice, numeric, etc.)